crates/burn-import/src/safetensors/serializer.rs

```diff
@@ -0,0 +1,700 @@
+use std::collections::HashMap;
+
+use burn::tensor::DType;
+use serde::{
+    Serialize,
+    ser::{SerializeSeq, SerializeStruct, Serializer},
+};
+
+#[derive(Debug)]
+pub struct SafetensorsTensorData {
+    bytes: Vec<u8>,
+    shape: Vec<usize>,
+    dtype: DType,
+}
+
+impl safetensors::View for SafetensorsTensorData {
+    fn dtype(&self) -> safetensors::Dtype {
```

Comment:
Can't create a view on TensorData directly without the need to save into `bytes`? This way there won't be an intermediate memory usage for large models. So if we have 8GB model sitting in GPU, we will not create a copy in RAM.